packages/mco/hooks/protected-applications.ts

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import { getNamespace } from '@odf/shared/selectors';
+import { useK8sWatchResources } from '@openshift-console/dynamic-plugin-sdk';
+import { RAMEN_PROTECTED_APPS_NAMESPACE } from '../constants';
+import { ACMPlacementModel } from '../models';
+import {
+  ACMSubscriptionKind,
+  ArgoApplicationSetKind,
+  DRPlacementControlKind,
+} from '../types';
+import { findPlacementNameFromAppSet, isPlacementModel } from '../utils';
+import {
+  getApplicationSetResourceObj,
+  getSubscriptionResourceObj,
+  getDRPlacementControlResourceObj,
+} from './mco-resources';
+
+const getPlacementUniqueName = (name, kind, namespace) =>
+  `${name}-${kind}-${namespace}`;
+
+const getResources = () => ({
+  applicationSets: getApplicationSetResourceObj(),
+  subscriptions: getSubscriptionResourceObj(),
+  drPlacementControls: getDRPlacementControlResourceObj(),
+});
+
+export const useApplicationsWatch: UseApplicationsWatch = () => {
+  const response = useK8sWatchResources<WatchResourceType>(getResources());
+
+  const {
+    data: appSets,
+    loaded: appSetsLoaded,
+    loadError: appSetsLoadError,
+  } = response?.applicationSets;
+
+  const {
+    data: subscriptions,
+    loaded: subscriptionsLoaded,
+    loadError: subscriptionsLoadError,
+  } = response?.subscriptions;
+
+  const {
+    data: drpcs,
+    loaded: drpcsLoaded,
+    loadError: drpcsLoadError,
+  } = response?.drPlacementControls;
+
+  const drpcLoadedWAError = drpcsLoaded && !drpcsLoadError;
+  const subsLoadedWAError = subscriptionsLoaded && !subscriptionsLoadError;
+  const appSetsLoadedWAError = appSetsLoaded && !appSetsLoadError;
+
+  const [protectedPlacementMap, drPolicyToDiscoveredAppsCount]: [
+    ProtectedPlacementMap,
+    DRPolicyToAppCount
+  ] = React.useMemo(() => {
+    const protectedPlacementMap: ProtectedPlacementMap = {};
+    const drPolicyToDiscoveredAppsCount: DRPolicyToAppCount = {};
+    if (drpcLoadedWAError) {
+      drpcs?.forEach((drpc) => {
+        const placementRef = drpc.spec.placementRef;
+        const namespace = getNamespace(drpc);
+        const placementUniqueName = getPlacementUniqueName(
+          placementRef.name,
+          placementRef.kind,
+          namespace
+        );
+
+        if (!protectedPlacementMap.hasOwnProperty(placementUniqueName)) {
+          protectedPlacementMap[placementUniqueName] = new Set();
+        }
+        protectedPlacementMap[placementUniqueName].add(drpc);
```

Comment:
```suggestion
        protectedPlacementMap[placementUniqueName] = drpc;
```

placement placementUniqueName and drpc is one to one mapping.